texar/data/data/multi_aligned_data.py

```diff
@@ -0,0 +1,701 @@
+                    vocab = vocabs[vocab_shr]
+                else:
+                    vocab = Vocab(hparams[vocab_shr]["vocab_file"],
+                                  bos_token=bos_token,
+                                  eos_token=eos_token)
+            else:
+                vocab = Vocab(hparams_i["vocab_file"],
+                              bos_token=bos_token,
+                              eos_token=eos_token)
+            vocabs.append(vocab)
+
+        return vocabs
+
+    @staticmethod
+    def make_embedding(hparams, vocabs):
+        r"""Optionally loads embeddings from files (if provided), and
+        returns respective :class:`texar.data.Embedding` instances.
+        """
+        if not isinstance(hparams, (list, tuple)):
+            hparams = [hparams]
+
+        embs = []
+        for i, hparams_i in enumerate(hparams):
+            if not _is_text_data(hparams_i["data_type"]):
+                embs.append(None)
+                continue
+
+            emb_shr = hparams_i["embedding_init_share_with"]
+            if emb_shr is not None:
+                if emb_shr >= i:
+                    MultiAlignedData._raise_sharing_error(
+                        i, emb_shr, "embedding_init_share_with")
+                if not embs[emb_shr]:
+                    raise ValueError("Cannot share embedding with dataset %d "
+                                     "which does not have an embedding." %
+                                     emb_shr)
+                if emb_shr != hparams_i["vocab_share_with"]:
+                    raise ValueError("'embedding_init_share_with' != "
+                                     "vocab_share_with. embedding_init can "
+                                     "be shared only when vocab is shared.")
+                emb = embs[emb_shr]
+            else:
+                emb = None
+                emb_file = hparams_i["embedding_init"]["file"]
+                if emb_file and emb_file != "":
+                    emb = Embedding(vocabs[i].token_to_id_map_py,
+                                    hparams_i["embedding_init"])
+            embs.append(emb)
+
+        return embs
+
+    def process(self, raw_example: Tuple[Union[str, Dict[str, Any]], ...]) \
+            -> Tuple[Union[List[str], int, float, Dict[str, Any]], ...]:
+        dataset_hparams = self.hparams.datasets
+        processed_examples = []
+        for i, raw_example_i in enumerate(raw_example):
+            if _is_text_data(dataset_hparams[i]["data_type"]):
+                delimiter = dataset_hparams[i]["delimiter"]
+                raw_tokens = raw_example_i.split(delimiter)  # type: ignore
+                max_seq_length = dataset_hparams[i]["max_seq_length"]
+                if (max_seq_length is not None and
+                        len(raw_tokens) > max_seq_length):
+                    if dataset_hparams[i]["length_filter_mode"] is \
+                            _LengthFilterMode.TRUNC.value:
```

Comment:
The `is` comparison only works when `dataset_hparams[i]["length_filter_mode"]` is a *string literal*, i.e. when the user says `"trunc"` in the code, but not in a config file that is read. Generally, use `==` for comparison unless you're sure comparing by `id` (pointer address) is fine.